src/Serializer/AbstractSerializer.php

```diff
@@ -241,7 +241,14 @@ protected function serializeValue($value)
         }
 
         if (\is_object($value)) {
-            return 'Object ' . \get_class($value);
+            $objectId = null;
+            if (isset($value->id)) {
+                $objectId = $value->id;
+            } elseif (\is_callable([$value, 'getId']) && method_exists($value, 'getId')) {
```

Comment:
Not sure there is any reason to call both is_callable() and method_exists() - if it's callable it must also exist? One hopes.

Reply:
According to https://www.php.net/manual/en/function.is-callable.php

> Notes:
> An object is always callable if it implements [__invoke()](https://www.php.net/manual/en/language.oop5.magic.php#object.invoke), and that method is visible in the current scope.
A class name is callable if it implements [__callStatic()](https://www.php.net/manual/en/language.oop5.overloading.php#object.callstatic).
If an object implements [__call()](https://www.php.net/manual/en/language.oop5.overloading.php#object.call), then this function will return true for any method on that object, even if the method is not defined.

So it makes sense to use both functions IMO: 
- `method_exists()` to check if the method `getId()` exists (it doesn't check if it's public or not)
- `is_callable()` to check if getId()` can be called

Reply:
Ok today I learned :)